mteb/deprecated_evaluator.py

```diff
@@ -306,13 +315,16 @@ def run(
         elif verbosity == 3:
             datasets.logging.set_verbosity(logging.DEBUG)
 
-        meta = self.create_model_meta(model)
-        output_path = self._create_output_folder(meta, output_folder)
-
+        mteb_model: MTEBModels
         if isinstance(model, SentenceTransformer):
-            model = SentenceTransformerEncoderWrapper(model)
+            mteb_model = cast(EncoderProtocol, SentenceTransformerEncoderWrapper(model))
```

Comment:
Minor nit, but generally prefer keeping casting operations separate from code changes (they are mostly there to appease the type checker and act as documentations so it feels weird to have them both intertwined)

Reply:
the cast is on the wrapper - you are stating that the wrapper is an EncoderProtocol (which it should be?)